enhancements/etcd/etcd-size-tuning.md

```diff
@@ -0,0 +1,230 @@
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - None
+creation-date: 2024-01-24
+last-updated: 2024-03-12
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/ETCD-514
+---
+
+# etcd Size Tuning
+
+## Summary
+
+This enhancement would replace the hardcoded value for the etcd backend database quota with a human readable, configurable value; 8GiB. This value does not immediately reserve disk space for etcd; it is an upper limit to the size of the etcd database on disk.
+This enhancement only covers the mvp for a tech preview release of this new feature; a future enhancement will be necessary.
+
+## Motivation
+
+Customers have asked for the ability to increase the quota for the backend database to faciliate larger clusters.
+Specifically, in our build clusters, the number of namespaces and CRDs (and the size of the CRDs) have been increasing which drastically increase the amount of data stored in etcd.
+In 4.14, the perf team is now unable to run their benchmark suite.
+We want to be able to allow for this while minimizing the risk of them setting values that cause issues.
+
+### User Stories
+
+* As an administrator, I want to change the etcd backend size to decrease the risk of quota reached errors in very large clusters.
+* As an Openshift support, I want to walk a customer through changing the etcd size in a minimal number of steps.
+* As an etcd engineer, I want to easily add and test new sizes for performance impacts on both large and small clusters.
+
+### Goals
+
+* Allow configuration of the backend limit via human readable units: 16GiB.
+* Add an API to allow admins to change the value.
+* The backend limit can only be increased and not decreased.
+
+### Non-Goals
+
+* Handle consuming value changes without an etcd rollout.
+
+## Proposal
+
+Allow the customer/admin access to set the backend quota directly through a human readable value via the etcd.operator.openshift.io/v1 techpreview field. The only restriction is that they are disallowed from setting the value lower than the current.
+
+Increasing this value will not have etcd immediately reserve this space on disk, it just allows etcd to grow up to that value if etcd grows under the usage of the cluster.
+
+The default value or unset value would be the current, hardcoded backend quota: 8GiB. This would allow for upgrades where this configuration value wouldn't nesscessarily be set.
+An initial upper limit of 32GiB will be enforced so as to limit the impact of some etcd operations, like restore, transporting snapshots, etc. 32GiB is an arbitrary value and work with the Perf-Scale team will need to be done to validate this value and potentially increase the upper limit; finding a reasonable and safe upper bound will need to be done before this feature is made GA.
```

Comment:
Based on the work in https://issues.redhat.com/browse/PERFSCALE-2881 this is not arbitrary number anymore. But rather a proven value, where our perf team tested up to 40G being the point where cluster was unstable and unrecoverable.